### DISCUSSION TITLE
Is it possible to define 'preset' and 'system_mode' for the same DataPoint

### DISCUSSION THREAD

**Discussion by TantrumV4** [Koenkk/zigbee2mqtt#24219] [Devices]
Hi, i bought some TRV from Aliexpress and I am trying to add a support for a new device following Z2M docs (it is some Tuya Zigbee TRV). 

On the DataPoint 2 the TRV have defined the presets/system_modes, like this:

```
[
        2,
        "preset",
        tuya.valueConverterBasic.lookup({
          off: tuya.enum(0),      // HA Off -> TRV Off
          antifrost: tuya.enum(1),  // Preset Antifrost
          eco: tuya.enum(2),        // Preset Eco
          comfort: tuya.enum(3),    // Preset Comfort
          auto: tuya.enum(4),     // HA Auto -> TRV Program
          heat: tuya.enum(5),     // HA Heat -> TRV On
        }),
        { readable: true, writable: true },
      ],
```
Why it is not possible to make something like this so the HA climate card can support both presets and system_modes on the same datapoint?

```
      [
        2,
        "system_mode",
        tuya.valueConverterBasic.lookup({
          off: tuya.enum(0),      // HA Off -> TRV Off
          auto: tuya.enum(4),     // HA Auto -> TRV Program
          heat: tuya.enum(5),     // HA Heat -> TRV On
        }),
        { readable: true, writable: true },
      ],
      [
        2,
        "preset",
        tuya.valueConverterBasic.lookup({
          antifrost: tuya.enum(1),  // Preset Antifrost
          eco: tuya.enum(2),        // Preset Eco
          comfort: tuya.enum(3),    // Preset Comfort
        }),
        { readable: true, writable: true },
      ],
```

I would like to have the presets like `antifrost`, `eco` and `comfort` available to use, but at the same time to be able to control the TRV with system_mode like `auto`, `heat` and `off`. In my head it seems like the presets should be pre-selecting the temperature and the system modes should be telling the device how to operate (fully open to heat, fully close to cool or manually select temp and maintaining the temperature).
![image](https://github.com/user-attachments/assets/1b54eee0-e83e-425b-b074-8cdb69dd4b5c)

**Comment by Koenkk** [comment]
This is possible through a custom value converter ([example](https://github.com/Koenkk/zigbee-herdsman-converters/blob/92441ff768f2ffaa4b301211e505854df1089bfc/src/lib/tuya.ts#L604))
